### PR TITLE
Add football plugin

### PR DESCRIPTION
**Pull request by eduardourso**
This PR adds the football plugin to the Novoda dashboard.

As the idea is to have a realtime updated plugin because of the matches, we did the creation of the UI dinamically in the `runtime.js` file.

While configuring the dashboard you should be able to follow the following list of competitions:

 Competition-ID | Country | Competition Name
-- | -- | --
452 | Germany | 1. Bundesliga
453 | Germany | 2. Bundesliga
445 | England | Premier League
447 | England | League On
446 | England | Championship
456 | Italy | Serie A
459 | Italy | Serie B
455 | Spain | Primera Division
450 | France | Ligue 1
451 | France | Ligue 2
449 | Netherlands | Eredivisie
457 | Portugal | Primeira Liga
464 | Europe | Champions-League
467 | World | World-Cup

There are a few other avaliable ones but I'll add to the plugin a way of selecting a competition from a list instead of using its ID. (if that's even possible)

 World Cup | Champions League |  extratime and penalties example
-- | -- | --
screen shot 2018-04-24 at 13 21 37 | screen shot 2018-04-24 at 13 22 21 | screen shot 2018-04-24 at 13 21 53

### CODE REVIEW

File: football/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "football-plugin",
+  "version": "1.0.5",
+  "main": "lib/index.js",
+  "repository": {},
+  "license": "MIT",
+  "dependencies": {
+    "dashboard-plugin": "https://github.com/novoda/dashboards/blob/master/plugin/dashboard-plugin-1.0.3.tgz?raw=true"
```
Comment (ouchadam) [inline football/package.json]: we need to use 1.0.7 :palm_tree:

File: football/lib/plugin.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    template: 'template.html',
+    __dirname
+}
+
+const configuration = () => {
+    return {
+        "football-plugin": {
+            name: 'football-plugin',
+            template: {
+                awayTeamName: plugin.createStringField('awayTeamName'),
```
Comment (ouchadam) [inline football/lib/plugin.js]: are all these fields needed in the dashboard configuration?

File: football/lib/flagMapper.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+const flagForTeam = (teamName) => {
```
Comment (ouchadam) [inline football/lib/flagMapper.js]: This file isn't used any more but are we keeping it for the future?

**Comment by tailec** [comment]
I think for England it should be this flag: https://emojipedia.org/flag-for-england/ :)

File: football/lib/flagMapper.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        case `Panama`:
+            return `🇵🇦`
+        case `Tunisia`:
+            return `🇹🇳`
+        case `Sweden`:
+            return `🇸🇪`
+        case `Poland`:
+            return `🇵🇱`
+        case `England`:
+            return `🇬🇧`
```
Comment (tailec) [inline football/lib/flagMapper.js]: I think for England it should be this flag: https://emojipedia.org/flag-for-england/ :)

File: football/lib/flagMapper.js (hunk 3 above)
Comment (eduardourso) [reply football/lib/flagMapper.js]: yes, once we find a way to use emojis in the runtime.js

File: football/lib/flagMapper.js (hunk 4 above)
Comment (ouchadam) [reply football/lib/flagMapper.js]: raised an issue on the main repo https://github.com/novoda/dashboards/issues/38

**Comment by blundell** [comment]
if we do a sweep for the world cup, is there a way to put the sweep persons name under this team?

So it'd show

`Urso vs Adam`

**Comment by ouchadam** [comment]
perhaps we could build a more dedicated plugin based on this one?

**Comment by eduardourso** [comment]
@blundell That'd be so cool, let's think about it.

**Pull request merged**